cpp/examples/tutorial_examples/run.sh

```diff
@@ -45,6 +45,8 @@ echo "== Running example project"
 echo "=="
 echo
 
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib
```

Comment:
@kou I am slightly confused on why if I don't add the `/usr/local/lib` to `LD_LIBRARY_PATH` the job fails with:
```
/build/example/dataset_example: error while loading shared libraries: libarrow_acero.so.1300: cannot open shared object file: No such file or directory
```
Any idea if I am missing something?

Reply:
Thanks @kou ! I had to add `CMAKE_PREFIX_PATH` too otherwise I was getting:
```
CMake Error at CMakeLists.txt:22 (find_package):
  By not providing "FindArrow.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Arrow", but
  CMake did not find one.

  Could not find a package configuration file provided by "Arrow" with any of
  the following names:

    ArrowConfig.cmake
    arrow-config.cmake

  Add the installation prefix of "Arrow" to CMAKE_PREFIX_PATH or set
  "Arrow_DIR" to a directory containing one of the above files.  If "Arrow"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
```
with all these changes it is still failing to find `libarrow_acero.so.1300` when executing.

Reply:
Oh, sorry. How about specifying rpath to Arrow C++ too?

```diff
diff --git a/cpp/examples/tutorial_examples/run.sh b/cpp/examples/tutorial_examples/run.sh
index 4e5e8eeab..bc0078435 100755
--- a/cpp/examples/tutorial_examples/run.sh
+++ b/cpp/examples/tutorial_examples/run.sh
@@ -30,7 +30,8 @@ echo "== Building Arrow C++ library"
 echo "=="
 echo
 
-ARROW_CMAKE_OPTIONS="-DCMAKE_INSTALL_PREFIX=${ARROW_INSTALL_DIR}" \
+ARROW_CMAKE_OPTIONS="-DCMAKE_INSTALL_PREFIX=${ARROW_INSTALL_DIR} \
+                     -DCMAKE_INSTALL_RPATH=${ARROW_INSTALL_DIR}/lib" \
   ./build_arrow.sh
 
 echo
@@ -39,7 +40,8 @@ echo "== Building example project using Arrow C++ library"
 echo "=="
 echo
 
-EXAMPLE_CMAKE_OPTIONS="-DCMAKE_INSTALL_RPATH=${ARROW_INSTALL_DIR}/lib -DCMAKE_PREFIX_PATH=${ARROW_INSTALL_DIR}" \
+EXAMPLE_CMAKE_OPTIONS="-DCMAKE_INSTALL_RPATH=${ARROW_INSTALL_DIR}/lib \
+                       -DCMAKE_PREFIX_PATH=${ARROW_INSTALL_DIR}" \
   ./build_example.sh
 
 echo
```

Reply:
Thanks @kou ! This worked locally, running on archery now.